bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/types/XSAnyURI.java

```diff
@@ -107,7 +110,17 @@ public ResultSequence constructor(ResultSequence arg) throws DynamicError {
 			throw DynamicError.invalidType();
 		}
 
-		return new XSAnyURI(aat.string_value().trim());
+		String uri = aat.string_value().trim().replaceAll("\\s+", " ");
+
+		try {
+			// The URI constructor does not allow raw spaces. Escape them to ensure they do not cause the xs:anyURI
+			// constructor to fail.
+			new URI(uri.replace(" ", "%20"));
```

Comment:
Will we have issues with other characters in the URI?  Should we be doing URI escaping in general, rather than just looking for spaces?